README.md

```diff
@@ -1,30 +1,74 @@
+You'll need a working [OpenStack installation](https://microstack.run/docs/single-node) with
+flavors with a minimum of 2 CPU cores, 8GB RAM and 10GB disk.
 
-## Testing
+### Set up
 
-Testing is run via tox and pytest. The unit test can be ran with `tox -e unit` and the integration test on juju 3.1 with `tox -e integration`.
+Follow [MicroStack's single-node](https://microstack.run/docs/single-node) starting guide to set 
+up MicroStack.
 
-Dependencies are installed in virtual environments. Integration testing requires a juju controller to execute. These tests will use the existing controller, creating an ephemeral model for the tests which is removed after testing. If you do not already have a controller setup, you can configure a local instance via LXD, see the [upstream documentation](https://juju.is/docs/lxd-cloud) for details.
+Follow the [tutorial on GitHub runner](https://charmhub.io/github-runner) to deploy the GitHub
+runner.
 
-## Generating src docs for every commit
+### Deploy
 
-Run the following command:
+Deploy the charm.
 
-```bash
-echo -e "tox -e src-docs\ngit add src-docs\n" >> .git/hooks/pre-commit
-chmod +x .git/hooks/pre-commit
 ```
+juju deploy github-runner-image-builder \
+--config experimental-external-build=True \
+--config experimental-external-build-network=<OPENSTACK-NETWORK-NAME> \
+--config openstack-auth-url=<OPENSTACK-AUTH-URL> \
+--config openstack-password=<OPENSTACK-PASSWORD> \
+--config openstack-project-domain-name=<OPENSTACK-PROJECT-DOMAIN-NAME> \
+--config openstack-project-name=<OPENSTACK-PROJECT-NAME> \
+--config openstack-user-name=<OPENSTACK-USER-NAME>
+
+juju integrate github-runner-image-builder github-runner
+```
+
+### Basic operations
+<!--Brief walkthrough of performing standard configurations or operations-->
+
+After having deployed and integrated the charm with the GitHub runner charm, the image should start
+to build and be provided to the GitHub runner automatically. The whole process takes around 10
+minutes.
+
+## Integrations
+<!-- Information about particularly relevant interfaces, endpoints or libraries related to the charm. For example, peer relation endpoints required by other charms for integration.--> 
+* image: The image relation provides the OpenStack image ID to the GitHub runners.
+* cos-agent: The COS agent subordinate charm provides observability using the Canonical
+Observability Stack (COS).
+
+For a full list of integrations, please refer to the [Charmhub documentation](https://charmhub.io/github-runner-image-builder/integrations).
+
+## Learn more
+* [Read more](https://charmhub.io/github-runner-image-builder) <!--Link to the charm's official documentation-->
+* [Developer documentation](https://github.com/canonical/github-runner-image-builder-operator/blob/main/CONTRIBUTING.md) <!--Link to any developer documentation-->
+* [Troubleshooting](https://matrix.to/#/!DYvOMMMjuXPZRJYHdy:ubuntu.com?via=ubuntu.com&via=matrix.org)
```

Comment:
Is it expected? Troubleshooting links to the Matrix room.

Reply:
Yes, this is what I've been telling people to link to if there's no official troubleshooting documentation/guide.

Reply:
Varshi pointed out on another PR that this is the wrong Matrix room, please update to https://matrix.to/#/#charmhub-charmdev:ubuntu.com